Assets/MixedRealityToolkit.Services/DiagnosticsSystem/MixedRealityDiagnosticsSystem.cs

```diff
@@ -264,7 +264,7 @@ public float FrameSampleRate
         /// <inheritdoc />
         public int GetHashCode(object obj) => SourceName.GetHashCode();
 
-        private void RaiseDiagnosticsChanged()
+        public void RaiseDiagnosticsChanged()
```

Comment:
I ended up exposing this in order to be able to use via tests (internal visibility wasn't sufficient) - thoughts? I could try to hide this if there's an allergic reaction to this.

Reply:
Not a huge fan of exposing things just for tests, but I don't think I see a huge risk with this change.

Reply:
Maybe make this protected and then create a TestMixedRealityDiagnosticsSystem.cs that shows a public interface to that protected method?

Reply:
Alrighty I took a look into a few options here:

1. Make protected and subclass for tests.
2. Use InternalsVisibleTo to have it at internal visibility.

I opted to go with option 2, even though from a cleanliness perspective, both feel roughly odd (i.e. subclassing and using a larger thing to escape a single function and adding that complexity to the test code + either doing a custom profile to instantiate it or new-ing it manually which is also different from how things are typically created). We also have some prior art for InternalsVisibleTo.

Reply:
I second using InternalsVisibleTo